src/eslint-runner/src/__tests__/shouldLintAll.test.js

```diff
@@ -10,14 +21,99 @@ test.each([
   '.eslintrc.json',
   'src/packages/relay/.eslintrc.js',
   '/Users/TEST/universe/src/packages/relay/.eslintrc.js',
-  'package.json',
-  'src/packages/relay/package.json',
-  '/Users/code/universe/src/packages/relay/package.json',
   '.eslintignore',
 ])('filename "%s" IS eslint config file', filename => {
   expect(shouldLintAll(filename)).toBe(true);
 });
 
+it('should lint all for package.json if dependency is removed', () => {
+  jest.spyOn(utils, 'ShellCommand').mockImplementationOnce(() => ({
+    runSynchronously: () => ({
+      getStdout: () => JSON.stringify({ dependencies: { a: 'lol', b: 'lol2' } }),
+    }),
+  }));
+  jest.doMock('../../../../package.json', () => ({
+    dependencies: { a: 'lol' },
+  }));
+  expect(shouldLintAll('package.json')).toBe(true);
+});
+
+it('should not lint all for package.json if dependency is added', () => {
+  jest.spyOn(utils, 'ShellCommand').mockImplementationOnce(() => ({
+    runSynchronously: () => ({
+      getStdout: () => JSON.stringify({ dependencies: { a: 'lol', b: 'lol2' } }),
+    }),
+  }));
+  jest.doMock('../../../../package.json', () => ({
+    dependencies: { a: 'lol', b: 'lol2', c: 'lol3' },
+  }));
+
+  expect(shouldLintAll('package.json')).toBe(false);
+});
+
+it('should not lint all for package.json if dependency is upgraded', () => {
```

Comment:
is this correct? what if I upgrade eslint? 🤔

Reply:
You can do that without even changing package.json

Reply:
that's true (e.g. only yarn.lock can be upgraded and some people might not use it), but still... upgrade in one eslint plugin can produce tons of issues which are discovered much later

Reply:
Yes, one way of diffing, is as @mrtnzlml also mentioned, check if `eslint-plugin-*` is changed. 

I don't think there is a 100% fool-proof way. Even today, you can upgrade eslint and not have the lint runner run lint any files so 🤷‍♂️

Reply:
if it's not an extra complication, I would add this behind some flag or a possibility to opt-out

Reply:
Yeah, I think I will just close this one. I think I can live with how it works today as well.